fml/task_source.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_TASK_SOURCE_H_
+#define FLUTTER_FML_TASK_SOURCE_H_
+
+#include "flutter/fml/delayed_task.h"
+#include "flutter/fml/task_queue_id.h"
+
+namespace fml {
+
+class MessageLoopTaskQueues;
+
+enum TaskSourceGrade {
+  kPrimary,
+  kSecondary,
+};
+
+class TaskSource {
+ public:
+  struct TopTask {
+    TaskQueueId task_queue_id;
+    TaskSourceGrade task_source_grade;
+    DelayedTask task;
+  };
+
+  explicit TaskSource(TaskQueueId task_queue_id);
+
+  ~TaskSource();
+
+  void Clear();
+
+  void RegisterTask(TaskSourceGrade grade, DelayedTask&& task);
```

Comment:
Can `DelayedTask` have its copy ctor disabled? We can then remove the && and ensure that it is impossible to have unnecessary copies.

Reply:
I tried doing this, looks like using a priority queue without a copy constructor is non-trivial. Will attempt doing this in a future PR.